main.cpp

```diff
@@ -14,7 +19,20 @@ int main(int argc, char** argv) {
   int ret = parser.parse();
 
   yylex_destroy();
+
+  // 1 if parsing failed because of invalid input;
+  // 2 if parsing failed due to memory exhaustion.
```

Comment:
Only found this in the manual [1]. Just curious where did you know the definition of return value.

> Method on parser: `int parse ()`
> Run the syntactic analysis, and return 0 on success, 1 otherwise. 

[1] https://www.gnu.org/software/bison/manual/html_node/C_002b_002b-Parser-Interface.html#index-parse-on-parser